spec/requests/todos_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'rails_helper'
+
+RSpec.describe "Todos API", type: :request do
+  let!(:todos) { create_list(:todo, 10) }
+  let(:todo_id) { todos.first.id }
+
+  describe "GET /todos" do
+    before { get "/todos" }
+
+    it "returns todos" do
+      expect(json).not_to be_empty
+      expect(json.size).to eq(10)
+    end
+
+    it "returns status code 200" do
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  describe "POST /todos" do
+    let(:valid_attributes) { { title: "Learn Rails", done: false }.to_json }
+
+    context "when the request is valid" do
+      before { post "/todos", params: valid_attributes, headers: { 'CONTENT_TYPE' => 'application/json' } }
+
+      it "creates a todo" do
+        expect(JSON.parse(response.body)['title']).to eq('Learn Rails')
+        #expect(json["title"]).to eq("Learn Rails")
```

Comment:
with `json` object, was it not working?

Reply:
no it wasnt working as the data response had to come from body